BepInEx.Core/Bootstrap/LoadingSystem.cs

```diff
@@ -174,56 +202,7 @@ protected static bool PluginTargetsWrongBepin(PluginInfo pluginInfo)
     /// <summary>
     ///     Occurs after all plugins are loaded.
     /// </summary>
-    public event Action Finished;
-
-    public virtual void Initialize(string gameExePath = null)
-    {
-        if (_initialized)
-            throw new InvalidOperationException("Chainloader cannot be initialized multiple times");
-
-        // Set vitals
-        if (gameExePath != null)
-            // Checking for null allows a more advanced initialization workflow, where the Paths class has been initialized before calling Chainloader.Initialize
-            // This is used by Preloader to use environment variables, for example
-            Paths.SetExecutablePath(gameExePath);
-
-        InitializeLoggers();
-
-        if (!Directory.Exists(Paths.PluginProviderPath))
-            Directory.CreateDirectory(Paths.PluginProviderPath);
-
-        if (!Directory.Exists(Paths.PluginPath))
-            Directory.CreateDirectory(Paths.PluginPath);
-
-        _initialized = true;
-
-        Logger.Log(LogLevel.Message, "Chainloader initialized");
-    }
-
-    protected virtual void InitializeLoggers()
-    {
-        if (ConsoleManager.ConsoleEnabled && !ConsoleManager.ConsoleActive)
-            ConsoleManager.CreateConsole();
-
-        if (ConsoleManager.ConsoleActive)
-        {
-            if (!Logger.Listeners.Any(x => x is ConsoleLogListener))
-                Logger.Listeners.Add(new ConsoleLogListener());
-
-            ConsoleManager.SetConsoleTitle(ConsoleTitle);
-        }
-
-        if (ConfigDiskLogging.Value)
-            Logger.Listeners.Add(new DiskLogListener("LogOutput.log", ConfigDiskLoggingDisplayedLevel.Value,
-                                                     ConfigDiskAppend.Value, ConfigDiskLoggingInstantFlushing.Value,
-                                                     ConfigDiskLoggingFileLimit.Value));
-
-        if (!TraceLogSource.IsListening)
-            Logger.Sources.Add(TraceLogSource.CreateSource());
-
-        if (!Logger.Sources.Any(x => x is HarmonyLogSource))
-            Logger.Sources.Add(new HarmonyLogSource());
-    }
+    public event Action AllPluginsLoaded;
```

Comment:
A progress event would be much more useful I think, with args saying what plugin is about to be loaded, was loaded, when loading started and then finished. Ideally the event would be static and fire for all LoadingSystems, just have the originating LoadingSystem in the args. It would make https://github.com/BepInEx/BepInEx.Debug/?tab=readme-ov-file#startup-profiler trivial.

Reply:
Changed events to use event args and be static, the progression part is something I am unsure, but it can be added later in case